test/functional/page_objects/common_page.js

```diff
@@ -151,11 +151,21 @@ export function CommonPageProvider({ getService, getPageObjects }) {
 
     navigateToApp(appName, { basePath = '', shouldLoginIfPrompted = true, shouldAcceptAlert = true, hash = '' } = {}) {
       const self = this;
-      const appConfig = config.get(['apps', appName]);
-      const appUrl = getUrl.noAuth(config.get('servers.kibana'), {
-        pathname: `${basePath}${appConfig.pathname}`,
-        hash: hash || appConfig.hash,
-      });
+
+      let appUrl;
```

Comment:
Can be compacted via ternary to:

```ts
const appConfig = config.get(['apps', appName]);
const appUrl = getUrl.noAuth(
  config.get('servers.kibana'),
  appConfig
    ? { pathname: `${basePath}${appConfig.pathname}`, hash: hash || appConfig.hash }
    : { pathname: `${basePath}/app/${appName}` }
);
```

Reply:
This doesn't work because `config.get(['apps', appName])` throws an exception if it doesn't exist.

Reply:
Aww, that's unfortunate.